src/server/components/upload-modal.jsx

```diff
@@ -0,0 +1,50 @@
+import React from 'react'
+import Modal from './modal'
+import ModalTitle from './modal-title'
+import ModalContent from './modal-content'
+import ModalCall from './modal-call'
+import { TextButton } from './buttons'
+import { fetchWithCSRFTokenAndJSONContent } from '../../shared/fetch-with-csrf-token'
+
+// delete this file
+// upload this file.
+
+function deleteFile(fileID) {
+  return fetchWithCSRFTokenAndJSONContent(`/api/v1/files/${fileID}`, { method: 'DELETE' })
+}
+
+export default class UploadModal extends React.Component {
+  constructor(props) {
+    super(props)
+    this.deleteAndReuploadFile = this.deleteAndReuploadFile.bind(this)
+  }
+
+  deleteAndReuploadFile() {
```

Comment:
I think this is not quite the right way of doing this. I think we should just upload the new file as a `PUT` request on the resource. That's just one step as opposed to two. Am I missing something?

Reply:
From my notes, it looked like we had agreed on this approach last week, which was why I went with this. My notes might have been off, since it seems obvious that `PUT` at this point is the right approach at this step, not `DELETE` + `POST`.

Reply:
Interesting, I wonder what I was thinking?

Reply:
I think it was just a misunderstanding on my part. If we know a file already exists on the server and a user assents to replacing it with the selected file, `PUT` seems like obviously the right approach. The UX layer that wraps the request was what we agreed on, which primarily takes care of the conflation issue (vs overloading `PUT` to always `POST` if the resource isn't there, or something like that).

Reply:
this all said, I'll switch to `PUT` for the modal resolution part.